src/main/java/com/monumental/util/csvparsing/CsvMonumentConverter.java

```diff
@@ -18,159 +19,174 @@
-     * Static method for converting a CSV row representing a Monument into a CsvMonumentConverterResult object
-     * @param csvRow - CSV row representing a Monument, as a String
-     * @param fromZipFile - True if the csvRow originated from a .zip file, False otherwise
-     * @return CsvMonumentConverterResult - the CsvMonumentConverterResult that is represented by the CSV row
-     */
-    public static CsvMonumentConverterResult convertCsvRow(String csvRow, boolean fromZipFile) {
-        if (csvRow == null) {
-            return null;
-        }
+    public static List<CsvMonumentConverterResult> convertCsvRows(List<String[]> csvRows, Map<String, String> mapping,
+                                                                  ZipFile zipFile) {
+        String[] headers = csvRows.get(0);
+        csvRows.remove(0);
+
+        Map<Integer, String> fields = CsvFileHelper.getFieldPositions(headers, mapping);
+        List<CsvMonumentConverterResult> results = new ArrayList<>();
+        for (String[] row : csvRows) {
+            List<String> values = Arrays.asList(row);
+            Monument monument = new Monument();
+            CsvMonumentConverterResult result = new CsvMonumentConverterResult();
+            Double latitude = null;
+            Double longitude = null;
+            for (int i = 0; i < values.size(); i++) {
+                String field = fields.get(i);
+                String value = values.get(i);
+                if (field == null || value.equals("")) continue;
+                switch (field) {
+                    case "submittedBy":
+                        Contribution contribution = parseContribution(value);
+                        contribution.setMonument(monument);
+                        monument.getContributions().add(contribution);
+                        break;
+                    case "artist":
+                        monument.setArtist(value);
+                        break;
+                    case "title":
+                        monument.setTitle(value);
+                        break;
+                    case "date":
+                        try {
+                            monument.setDate(parseDate(value));
+                        } catch (Exception e) {
+                            result.getWarnings().add("Date should be a valid date in the format DD-MM-YYYY or YYYY.");
+                        }
+                        break;
+                    case "materials":
+                        result.getMaterialNames().addAll(parseTags(value));
+                        break;
+                    case "inscription":
+                        monument.setInscription(value);
+                        break;
+                    case "latitude":
+                        try {
+                            latitude = Double.parseDouble(value);
+                        } catch (NumberFormatException e) {
+                            result.getWarnings().add("Latitude should be a valid number.");
```

Comment:
Reminder of our conversation about making this a warning and an error.

Reply:
I think we decided to leave this, right? Because as a warning, it allows you to still insert if you have a valid address whereas if I make it an error then you would not be able to. It becomes a warning if you have no address since then you have no coordinates or address

Reply:
Yeah, you're right.